samza-sql/src/main/java/org/apache/samza/sql/interfaces/SourceResolver.java

```diff
@@ -19,26 +19,21 @@
 
 package org.apache.samza.sql.interfaces;
 
+
 /**
- * Source Resolvers are used by Samza Sql application to fetch the {@link SqlSystemSourceConfig} corresponding to the source.
+ * Source Resolvers are used by Samza Sql application to fetch the {@link SqlSystemSourceConfig} corresponding
+ * to the source. Additionally, it provides capability to resolve Samza table sources and creating the matching
+ * table descriptors.
  */
 public interface SourceResolver {
```

Comment:
I think we probably need to rename this to IOResolver right?

Reply:
Was thinking renaming could touch too many files. But given this PR is adding the write-path to SQL so it makes sense to do it in one go. I'll do the renaming.